### ISSUE TITLE
aws-load-balancer-controller pod using IMDSv1

### ISSUE THREAD

**Issue by CoatedMoose** [kubernetes-sigs/aws-load-balancer-controller#3321]
**Describe the bug**
The aws load balancer controller pod seems to be using AWS IMDSv1, despite v2 being available on the hosts.

As measured by the CloudWatch search `SEARCH('{AWS/EC2,AutoScalingGroupName} MetricName=MetadataNoToken', 'Sum', 60)`.

- 1 time/24h, 2 requests to IMDSv1 are made (~0600 UTC in my case)
- Any time the `aws-load-balancer-controller-xxxx-xxxx` pod is deleted

**Steps to reproduce**

- Deploy the controller (using `v2_5_4_full.yaml`)
- Delete a pod - the `Deployment` will replace it automatically (or allow it to run for up to 24h)
- Observe the metric reports requests with  `MetadataNoToken`.

**Expected outcome**

No requests reported.

**Environment**

* v2.5.3, v2.5.4
* Server version: v1.24.15-eks-a5565ad
* Using EKS (yes/no), if so version? yes

**Additional Context**:

I'm suspicious that this issue might be resolved by [this change](https://github.com/kubernetes-sigs/aws-load-balancer-controller/commit/d1b8fbb0bc6b8b7639e0be06cfd6693751ac604d). The version bump includes the [this change to aws-sdk-go (v1.44.211)](https://github.com/aws/aws-sdk-go/pull/4742/files#diff-1082fdae378a5abc33332afae3c56453249b51a8309b4e3d15278e8c23471d2f), which mentions

> This release allows IMDS support to be set to v2-only on an existing AMI, so that all future instances launched from that AMI will use IMDSv2 by default.

However that change is not yet available in the provided image. When will the next version (that includes that change) be available to test?

**Comment by CoatedMoose** [comment]
I see the new version was made available a few hours ago. I've deployed it, and it did not stop usages of IMDSv1.

**Comment by CoatedMoose** [comment]
I am pretty sure I've tracked the issue down. In my test environment, I updated the tokens configuration to required and the hop limit to scream-test to shake out any usages. There was no errors reported AFAICT.

In the production environment, I didn't make any changes, which meant the hop limit for the tokens was still set to 1. This would make IMDSv2 fail, and (presumably) `aws-load-balancer-controller` (or something it makes requests to) falls back to IMDSv1.

I've updated the hop limit to 2, and the metric to track has stayed at 0.
